### PR TITLE
feat(private-artworks): Add private artwork layout (isUnlisted) on artwork screen

### PR DESCRIPTION
**Pull request by damassi**
This PR resolves [AMBER-531]

### Description

This implements the new private artwork layout on the arwork/id page, which conditionally renders sections of the page if certain conditions are true. 

Due to how the page is laid out, this code is unavoidably ugly. We will have to return to this if there's ever a third type of layout. Along the way had to make very minor changes to a bunch of preexisting components, in order to hide certain things. Code quality has most certainly taken a hit, but there's little we can do. 

https://github.com/artsy/eigen/assets/236943/7a0eafdc-c7e0-49a2-9152-46f800020615

cc @artsy/amber-devs 

### PR Checklist

- [ ] I have tested my changes on **iOS** and **Android**.
- [ ] I hid my changes behind a **[feature flag]**, or they don't need one.
- [ ] I have included **screenshots** or **videos**, or I have not changed the UI.
- [ ] I have added **tests**, or my changes don't require any.
- [ ] I added an **[app state migration]**, or my changes do not require one.
- [ ] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [ ] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- Add support for `isUnlisted` (Private) artworks

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[AMBER-531]: https://artsyproduct.atlassian.net/browse/AMBER-531?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/app/Scenes/Artwork/Components/ArtworkHeader.tests.tsx [hunk 1]
```diff
@@ -40,37 +39,4 @@ describe("ArtworkHeader", () => {
     // image carousel component
     expect(screen.UNSAFE_queryAllByType(ImageCarousel)).toHaveLength(1)
   })
-
```
Comment (jpotts244) [inline src/app/Scenes/Artwork/Components/ArtworkHeader.tests.tsx:43]: 🙏🏻 thanks for this

File: src/app/Scenes/Artwork/Components/PrivateArtwork/PrivateArtworkExclusiveAccess.tsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    artwork
+  )
+
+  if (!data.isUnlisted) {
+    return null
+  }
+
+  const handleLinkPress = () => {
+    navigate(`/partner/${data.partner?.slug}`)
+  }
```
Comment (jpotts244) [inline src/app/Scenes/Artwork/Components/PrivateArtwork/PrivateArtworkExclusiveAccess.tsx:34]: [Total nitpick/non blocking]: I think there was some work that went into force to not display this link [if the partner does not have a publicly visible profile](https://github.com/artsy/force/pull/13780/files)

Just to raise, kinda edge casey!

Comment (damassi) [reply src/app/Scenes/Artwork/Components/PrivateArtwork/PrivateArtworkExclusiveAccess.tsx:34]: Ooh, nice catch, ty. Will fix 👍

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  ❌ **An error occurred while validating your changelog, please make sure you provided a valid changelog.**
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 9f63243605db4ce464767b6be0eb2355ec4d056f
</p>

File: src/app/Scenes/Artwork/Components/AboutArtist.tsx [hunk 3]
```diff
@@ -15,18 +15,26 @@ export const AboutArtist: React.FC<AboutArtistProps> = ({ artwork }) => {
 
   const hasSingleArtist = artists && artists.length === 1
 
-  const text =
+  const biographyBlurb =
     hasSingleArtist && artists[0]?.biographyBlurb?.text ? artists[0]?.biographyBlurb?.text : null
+
+  const partnerBiographyBlurb = artwork.artist?.partnerBiographyBlurb?.text
```
Comment (jpotts244) [inline src/app/Scenes/Artwork/Components/AboutArtist.tsx:21]: Phew, youre too fast! 🎖️

File: src/app/Scenes/Artwork/Components/AboutArtist.tsx [hunk 4]
```diff
@@ -67,8 +77,15 @@ export const AboutArtistFragmentContainer = createFragmentContainer(AboutArtist,
         biographyBlurb {
           text
         }
+
         ...ArtistListItem_artist
       }
+      artist {
+        partnerBiographyBlurb {
```
Comment (jpotts244) [inline src/app/Scenes/Artwork/Components/AboutArtist.tsx:84]: 🔥 🔥

File: src/app/Scenes/Artwork/Artwork.tests.tsx [hunk 5]
```diff
@@ -90,11 +90,19 @@ describe("Artwork", () => {
     renderWithWrappers(<TestRenderer />)
 
     // ArtworkAboveTheFoldQuery
-    resolveMostRecentRelayOperation(environment)
+    resolveMostRecentRelayOperation(environment, {
+      Artwork: () => ({
+        isUnlisted: false,
```
Comment (jpotts244) [inline src/app/Scenes/Artwork/Artwork.tests.tsx:95]: Oh I am so sorry 😬

**Review by jpotts244 — DISMISSED** [review]
✨

File: src/app/Scenes/Artwork/Components/PrivateArtwork/PrivateArtworkExclusiveAccess.tsx (hunk 2 above)
Comment (damassi) [reply src/app/Scenes/Artwork/Components/PrivateArtwork/PrivateArtworkExclusiveAccess.tsx:37]: Fixed in [here](https://github.com/artsy/eigen/pull/10129/commits/67cbfcdf89e256fc34c5d5b60537fb3737a52e78) -- did what y'all did over in force, where we render just bare text

**Pull request merged**